website/python/brewmonitor/views.py

```diff
@@ -1,22 +1,72 @@
-from flask import Blueprint, url_for
-from flask_mako import render_template
-
-home_bp = Blueprint(
-    'home',
-    __name__,
-    template_folder='../../../templates',
-)
-
-
-@home_bp.route('/')
-def index():
-    return render_template(
-        'home.html.mako'
-    )
-
-
-@home_bp.route('/login')
-def login():
-    # TODO(tr) implement using flask-login
-    # TODO(tr) Handle the next parameter so we can redirect to the appropriate page.
-    return ''
+from flask import Blueprint, url_for, request, redirect
+from flask_mako import render_template
+
+from flask_login import LoginManager, UserMixin, login_required, login_user, logout_user, current_user
+from brewmonitor.user import User
+
+home_bp = Blueprint(
+    'home',
+    __name__,
+    template_folder='../../../templates',
+)
+
+
+@home_bp.route('/')
+def index():
+    return render_template(
+        'home.html.mako'
+    )
+
+@home_bp.route('/login', methods=['GET', 'POST'])
+def login():
+        
+    if request.method == 'GET':
+        return render_template('login.html.mako')
+
+    else:
+        username = request.form.get('username')
+        password = request.form.get('password')
+        remember = True if request.form.get('remember') else False
+        
+        id = User.verify(username, password)
+        if id is not None:
+            login_user(id, remember=remember)
+            return redirect(url_for('home.index'))
+
+        # if the above check passes, then we know the user has the right credentials
+        return redirect(url_for('home.login')) # if user doesn't exist or password is wrong, reload the page
```

Comment:
I would even split this in 2 methods:

```
@home_bp.route('/login', methods=['GET'])
def login():
        return render_template('login.html.mako')

@home_bp.route('/login', methods=['POST'])
def check_login():
        username = request.form.get('username')
        password = request.form.get('password')
        remember = True if request.form.get('remember') else False

        id = User.verify(username, password)
        if id is not None:
            login_user(id, remember=remember)
            return redirect(url_for('home.index'))

        # if the above check passes, then we know the user has the right credentials
        return redirect(url_for('home.login')) # if user doesn't exist or password is wrong, reload the page
```

I will need to check how we can implement a next parameter to redirect automatically where we were coming from.

Reply:
Agreed, and pushed.
About the next, it is provided as a request.next or something like that, but I choose not to implement it to hide as much as possible the existing commands. Also it would require some checking to make sure the next is not an external url, and I couldn't be bother to implement the check.